File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;
class Solution {
    public int[] solution(int[] progresses, int[] speeds) {        
        int[] arr = new int[progresses.length];
        Stack<Integer> stack = new Stack<>();
        for(int i=0; i<progresses.length; i++) {
            // 작업하는데 걸리는 시간 (100이 넘으면 끝)
            int time = 1;
            while(true) {
                int work = progresses[i] + (speeds[i] * time);
                if(work >= 100) break;
                time++;
            }
            arr[i] = time;
        }
        List<Integer> list = new ArrayList<>();
        int cnt = 1;
        for(int i=0; i<arr.length; i++) {
            if(stack.isEmpty()) {
                stack.push(arr[i]);
            } else {
                if(stack.peek() >= arr[i]) {
                   cnt++; 
                } else {
                    stack.push(arr[i]);
                    list.add(cnt);
                    cnt = 1;
                }
            }
        }
        list.add(cnt);
        
        int[] res = new int[list.size()];
        for(int i=0; i<list.size(); i++) {
            res[i] = list.get(i);
        }
        
        return res;
    }
}

File: 프로그래머스/2/42586. 기능개발/기능개발.cpp
#include <string>
#include <vector>
#include <iostream>
using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    vector<int> workSpeed;
    int cnt = 1;
    for(int i=0; i<progresses.size(); i++) {
        int n = 1;
        while(true) {
            if(((100 - progresses[i]) - (speeds[i] * n)) <= 0) {
                workSpeed.push_back(n);
                break;
            }
            n++;
        }
    }
    int pivot = workSpeed[0];
    for(int i=1; i<workSpeed.size(); i++) {
        // 5 10 1 1 20 1 
        if(pivot < workSpeed[i]) {
            answer.push_back(cnt);
            pivot = workSpeed[i];
            cnt = 1;
        } else {
            cnt++;
        }
    }
    answer.push_back(cnt);
    // for(int i=0; i<workSpeed.size(); i++) {
    //     cout << workSpeed[i] << " ";
    // }
    return answer;
}